Lib/collections/__init__.py

```diff
@@ -979,6 +979,26 @@ def __contains__(self, key):
 
     # Now, add the methods in dicts but not in MutableMapping
     def __repr__(self): return repr(self.data)
+
+    def __or__(self, other):
+        if isinstance(other, UserDict):
+            return self.__class__(self.data | other.data)
+        if isinstance(other, type(self.data)):
```

Comment:
Is not `type(self.data)` always `dict`?

Reply:
Correct, I don't see a way it could be otherwise except if someone explicitly overwrote the `data` attribute. The docs describe it as a "real dictionary": https://docs.python.org/2/library/userdict.html#UserDict.IterableUserDict.data